src/solvers/sat/external_sat.cpp

```diff
@@ -0,0 +1,188 @@
+/*******************************************************************\
+
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+\*******************************************************************/
+
+#include "external_sat.h"
+
+#include "dimacs_cnf.h"
+
+#include <util/exception_utils.h>
+#include <util/run.h>
+#include <util/string_utils.h>
+#include <util/tempfile.h>
+
+#include <json/json_parser.h>
+
+#include <chrono>
+#include <cstdlib>
+#include <fstream>
+#include <random>
+#include <sstream>
+#include <string>
+#include <thread>
+
+class missing_configurationt : public cprover_exception_baset
+{
+public:
+  missing_configurationt(std::string message) : _message(std::move(message))
+  {
+  }
+
+  std::string what() const override
+  {
+    return _message;
+  }
+
+protected:
+  std::string _message;
+};
+
+external_satt::external_satt(message_handlert &message_handler, std::string cmd)
+  : cnf_clause_list_assignmentt(message_handler), _cmd(std::move(cmd))
+{
+}
+
+const std::string external_satt::solver_text()
+{
+  return "External SAT solver";
+}
+
+bool external_satt::is_in_conflict(literalt) const
+{
+  UNIMPLEMENTED;
```

Comment:
The original intent of `UNIMPLEMENTED` was to document incomplete parts of the code.  Ideally we weren't going to add more.  That said, it is not clear to me what would be better here.